saleor/payment/gateways/stripe/__init__.py

```diff
@@ -0,0 +1,233 @@
+        response = dict()
+
+    # Create transaction
+    txn = _create_transaction(
+        payment=payment,
+        amount=amount,
+        kind=TransactionKind.REFUND,
+        response=response)
+
+    return txn, error
+
+
+def void(payment, **connection_params):
+    client, error = _get_client(**connection_params), ''
+
+    capture_txn = payment.transactions.filter(
+        kind=TransactionKind.CHARGE, is_success=True).first()
+
+    if capture_txn is not None:
+        try:
+            # Retrieve stripe charge and refund all
+            stripe_charge = client.Charge.retrieve(capture_txn.token)
+            response = client.Refund.create(charge=stripe_charge.id)
+        except stripe.error.StripeError as exc:
+            response = _get_error_response_from_exc(exc)
+            error = exc.user_message
+    else:
+        error = errors.ORDER_NOT_CHARGED
+        response = dict()
+
+    # Create transaction
+    txn = _create_transaction(
+        payment=payment,
+        amount=payment.total,
+        kind=TransactionKind.VOID,
+        response=response)
+
+    return txn, error
+
+
+def get_form_class():
+    return StripePaymentModalForm
+
+
+def _get_client(**connection_params):
+    stripe.api_key = connection_params.get('secret_key')
+    return stripe
+
+
+def _create_stripe_charge(client, payment, amount, payment_token, capture):
+    """Create a charge with specific amount, ignoring payment's total."""
+    # Get currency
+    currency = str(payment.currency).upper()
+
+    # Get appropriate amount for stripe
+    stripe_amount = get_amount_for_stripe(amount, currency)
+
+    # Get billing name from payment
+    name = '%s %s' % (
+        payment.billing_last_name, payment.billing_first_name)
+
+    # Update shipping address to prevent fraud in Stripe
+    shipping = dict(name=name, address=dict(
+        line1=payment.order.shipping_address.street_address_1,
+        line2=payment.order.shipping_address.street_address_2,
+        city=payment.order.shipping_address.city,
+        state=payment.order.shipping_address.country_area,
+        country=dict(countries).get(
+            payment.order.shipping_address.country, ''),
+        postal_code=payment.order.shipping_address.postal_code,
+    ))
+
+    # Create stripe charge
+    stripe_charge = client.Charge.create(
+        capture=capture,
+        amount=stripe_amount,
+        currency=currency,
+        source=payment_token,
+        shipping=shipping,
+        description=name)
+
+    return stripe_charge
+
+
+def _create_transaction(payment, amount, kind, response):
+    # Get currency from response or payment
+    currency = response.get('currency', payment.currency)
+
+    # Get amount from response or payment
+    if 'amount' in response:
+        amount = get_amount_from_stripe(response.get('amount'), currency)
+
+    # Get token, which is an empty string for error responses
+    token = response.get('id', '')
+
+    # Check is_uccess from response
```

Comment:
Typo, and maybe we could reword to this?
```suggestion
    # Check if the response's status is flagged as succeeded
```

Reply:
Thanks, sounds better!